tasks/prepare-fbc-release/tests/test-prepare-fbc-release.yaml

```diff
@@ -0,0 +1,146 @@
+                                  "revision": "foo",
+                                  "url": "https://github.com/foo/bar"
+                              }
+                          },
+                          "repository": "test/foo/bar"
+                      },
+                      {
+                       "containerImage": "quay.io/hacbs-release-tests/test-ocp-version/test-fbc-component@sha256:f6e744662e342c1321deddb92469b55197002717a15f8c0b1bb2d9440aac2297",
+                          "name": "test-container-boo",
+                          "source": {
+                              "git": {
+                                  "context": "./",
+                                  "dockerfileUrl": "build/Dockerfile",
+                                  "revision": "foo",
+                                  "url": "https://github.com/foo/bar"
+                              }
+                          },
+                          "repository": "test/foo/bar"
+                      }
+                      ]
+              }
+              EOF
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+    - name: run-task
+      taskRef:
+        name: prepare-fbc-release
+      params:
+        - name: snapshotPath
+          value: snapshot_spec.json
+        - name: dataPath
+          value: data.json
+      runAfter:
+        - setup
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+    - name: check-result
+      params:
+        - name: snapshotPath
+          value: snapshot_spec.json
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - run-task
+      taskSpec:
+        workspaces:
+          - name: data
+        params:
+          - name: snapshotPath
+            type: string
+        steps:
+          - name: check-result
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env bash
+              set -eu
+
+              SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+              # Check if the snapshot_spec.json file exists
+              if [ ! -f "$SNAPSHOT_PATH" ]; then
+                  echo "Error: snapshot_spec.json does not exist."
+                  exit 1
+              fi
+
+              # Function to check for missing field
+              check_field() {
+                  local field_name=$1
+                  local field_value=$2
+                  local component_name=$3
+
+                  if [ "$field_value" == "null" ]; then
+                      echo "Error: '$field_name' not found in component '$component_name'."
+                      exit 1
+                  fi
+              }
+
+              # Read and iterate over components
+              jq -c '.components[]' "$SNAPSHOT_PATH" | while read -r component; do
```

Comment:
same as the first `while` vs `for` comment.

Reply:
Updated, thanks!